pep517/wrappers.py

```diff
@@ -41,15 +49,45 @@ def default_subprocess_runner(cmd, cwd=None, extra_environ=None):
     check_call(cmd, cwd=cwd, env=env)
 
 
+def norm_and_check(source_tree, requested):
+    """Normalise and check a backend path.
+
+    Ensure that the requested backend path is specified as a relative path,
+    and resolves to a location under the given source tree.
+
+    Return an absolute version of the requested path.
+    """
+    if os.path.isabs(requested):
+        raise ValueError("paths must be relative")
+
+    abs_source = os.path.abspath(source_tree)
+    abs_requested = os.path.abspath(os.path.join(abs_source, requested))
```

Comment:
After joining to `abs_source`, this should already be absolute in all cases, shouldn't it? Maybe `normpath` makes it clearer what is happening.

Reply:
True. Will fix.